skbio/alignment/_tabular_msa.py

```diff
@@ -685,13 +674,14 @@ def reindex(self, key=None, keys=None):
         array(['a', 'b'], dtype=object)
 
         """
-        if key is not None and keys is not None:
+        if minter is not None and keys is not None:
             raise ValueError(
-                "Cannot use both `key` and `keys` at the same time.")
+                "Cannot use both `minter` and `keys` at the same time.")
 
         keys_ = None
-        if key is not None:
-            keys_ = [resolve_key(seq, key) for seq in self._seqs]
+        if minter is not None:
```

Comment:
There are a couple of border cases to consider:

- a `TabularMSA` with a minter, calling `reindex` with `keys` should set new keys and remove the minter
- a `TabularMSA` with a minter, calling `reindex` without passing `keys` or `minter` should remove the keys *and* the minter

Can you modify the logic here and add unit tests for these cases?